tests/queries/0_stateless/03231_alter_no_properties_before_remove_modify_reset.sql

```diff
@@ -0,0 +1,157 @@
+DROP TABLE IF EXISTS a SYNC;
+CREATE TABLE a (x Int64, y Int64 MATERIALIZED 1 SETTINGS (max_compress_block_size = 30000)) ENGINE = MergeTree ORDER BY x;
+
+
+SELECT 'REMOVE';
+ALTER TABLE a MODIFY COLUMN y Int64 REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR }
```

Comment:
I think fuzzers failed because of some syntactically invalid queries. I remove the one that fuzzer were complaining about, but if they still fail, then I will just remove the client side tests. That it not good, because the strings in the server side checks are not fuzzed as these queries.

Reply:
You should not remove queries and instead you should fix the fuzzer. Most often it's an issue with the AST classes (like not cloning all the nodes) on clone.

Reply:
So I understood what is happening and it is quite tricky. Let's consider the following queries:
```
ALTER TABLE a MODIFY COLUMN y MATERIALIZED 3 REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 1
ALTER TABLE a MODIFY COLUMN y EPHEMERAL 4 REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 2
ALTER TABLE a MODIFY COLUMN y COMMENT 5 REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 3
ALTER TABLE a MODIFY COLUMN y CODEC(ZSTD) REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 4
```

All of these are fine on server side, but on client side query 3 doesn't result in an exception of `SYNTAX_ERROR`, but only a failed parsing by `parseQuery` returning no query.

Let's see in details what happens! We parse these one easily:
```
ALTER TABLE a MODIFY COLUMN y MATERIALIZED 3 REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 1
ALTER TABLE a MODIFY COLUMN y EPHEMERAL 4 REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 2
```
In both cases the parser throws an exception of `SYNTAX_ERROR`, everything is fine. However the next line is the interesting:
```
ALTER TABLE a MODIFY COLUMN y COMMENT 5 REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 3
```

Here the parser fails parsing `5` as a comment, because it is not a string literal. It doesn't throw, just doesn't return a valid query, so we [ignore](https://github.com/ClickHouse/ClickHouse/blob/64d463bb09d2084d22859b0b76735bb3273c700e/src/Client/ClientBase.cpp#L2095-L2104) the the rest of the query up until (and including) `;`, so the new value of `this_query_begin` will point right before the comment. In the next call of [`analyzeMultiQueryText`](https://github.com/ClickHouse/ClickHouse/blob/5d3d2c213965fe785bd7236d67b80e632b5ad5c8/src/Client/ClientBase.cpp#L2080) we set `this_query_end` to the same position as `this_query_begin` and we continue parsing this:
```
-- { clientError SYNTAX_ERROR } 3
ALTER TABLE a MODIFY COLUMN y CODEC(ZSTD) REMOVE MATERIALIZED; -- { clientError SYNTAX_ERROR } 4
```
The parser throws an exception of `SYNTAX_ERROR`, so neither `this_query_end` nor `this_query_begin` will be moved. Then we try to parse the hint from `this_query_end` to the end of that line, so only ` -- { clientError SYNTAX_ERROR } 3`, but that is not a valid hint as hints [cannot be leading comments](https://github.com/ClickHouse/ClickHouse/blob/5d3d2c213965fe785bd7236d67b80e632b5ad5c8/src/Client/TestHint.cpp#L133).

Therefore we don't expect syntax error, thus we [rethrow](https://github.com/ClickHouse/ClickHouse/blob/5d3d2c213965fe785bd7236d67b80e632b5ad5c8/src/Client/ClientBase.cpp#L2249-L2255) the exception.

Reply:
I somehow missed some queries that were not good yesterday, removed them in fb19005968bd4388110bc7f14ffe66a2390b63ba.